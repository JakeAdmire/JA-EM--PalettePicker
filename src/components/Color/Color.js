import React from 'react';

export function Color({hex}) {

  let style = { backgroundColor: hex };

  return (
    <div style={style} className="Color">
      { hex }
    </div>
  )
}