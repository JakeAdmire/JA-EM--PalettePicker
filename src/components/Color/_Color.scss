.Color {
  width: 18%; height: 100%;
}