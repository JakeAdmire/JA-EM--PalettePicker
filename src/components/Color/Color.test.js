import React from 'react';
import ReactDOM from 'react-dom';
import { Color } from './Color';
import { shallow } from 'enzyme';

describe('Color', () => {

  let wrapper;
  let mockHex = '#FFFFFF';
  let mockLocked = false;
  let mockToggleLocked = jest.fn();

  beforeEach(() => {
    wrapper = shallow(
      <Color  hex={mockHex} 
              locked={mockLocked} 
              toggleLocked={mockToggleLocked}/>
    )
  })

  it('matches the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })

  it('should invoke toggleLocked prop on click', () => {    
    wrapper.find('.Color').simulate('mouseDown');

    expect(mockToggleLocked).toHaveBeenCalledWith(mockHex);
  })

  describe('determineLightLevel(hex)', () => {

    it('should invoke determineLightLevel on component mount', () => {
      const mockSpy = jest.spyOn(wrapper.instance(), 'determineLightLevel');

      wrapper.instance().determineLightLevel(mockHex);

      expect(mockSpy).toHaveBeenCalledWith(mockHex);
    })

    it('should return a false boolean when hex parameter is light', () => {
      mockHex = '#FFFFFF'; // white

      const determineLightLevelResult = wrapper.instance().determineLightLevel(mockHex);

      expect(determineLightLevelResult).toEqual(false);
    })

    it('should return a true boolean when hex parameter is dark', () => {
      mockHex = '#000000'; // black

      const determineLightLevelResult = wrapper.instance().determineLightLevel(mockHex);

      expect(determineLightLevelResult).toEqual(true);
    })

  })

  describe('hexToName(hex)', () => {

    it('should invoke hexToName on component mount', () => {
      const mockSpy = jest.spyOn(wrapper.instance(), 'hexToName');

      wrapper.instance().hexToName(mockHex);

      expect(mockSpy).toHaveBeenCalledWith(mockHex);
    })

    it('should return a string based on the hex parameter', () => {
      const hexToNameResult = wrapper.instance().hexToName('#249DE2');

      expect(hexToNameResult).toEqual('Curious Blue'); 
    })

  })

})