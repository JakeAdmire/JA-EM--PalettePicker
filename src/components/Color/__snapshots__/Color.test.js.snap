// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Color matches the snapshot 1`] = `ShallowWrapper {}`;
